### ISSUE TITLE
Current version not infering entities in a string without whitespaces, ignoring the different tokens provided

### ISSUE THREAD

**Issue by JBrettas** [RasaHQ/rasa#5995]
**Rasa version**: 
1.10.2

**Python version**: 
3.7.3

**Operating system** (windows, osx, ...): 
Windows 10

**Issue**:
NLU models now don't infer correctly two entities that are part of a same string but without whitespaces, even if I guarantee that the two entities are two distinct tokens. This makes NER tasks less flexible and it was something that it was working fine on Rasa 1.9.7 (I did a downgrade to have this back working).

For example, the model can't infer a currency entity with value '$' if the sentence is something like 'it was $50'. Also, it can't infer month (MM) and day (DD) in separate entities if the date is presented as in '11-19-2020'. These patterns work just fine on Rasa 1.9.7.

Spacy tokenizer does creates separate tokens for the '-' character, as can be checked in the following Python code:
```py
import spacy
nlp = spacy.load('en_core_web_md')
doc = nlp('it was on 10-05-2020')

for i in range(len(doc)):
    print(doc[i])
```

The problem on the bot can be reproduced on rasa shell using the following nlu.md for training:
```md
## intent:inform_date
- its [10](MM)-[05](DD)
- today is [05](DD) of [january](month)
- yes it was [7](DD) of [june](month)
- [15](DD) of [december](month)
- it was [01](MM)-[02](DD)
- I did it at [10](MM)-[25](DD)-[2019](YYYY)
- it was [05](MM)-[01](DD)-[2020](YYYY)
- [02](MM)-[03](DD)-[2020](YYYY)

## intent:inform_value
- [$](currency)[40](value)
- [$](currency)[120](value)
- [$](currency) [10](value)
- [500.00](value)
- [20.00](value)
- it was [$](currency)[40.50](value)
```

**Error (including full traceback)**:
```
On rasa shell nlu, enter:
'it was $50.00'

There will be no entities infered for '$' and '50.00'.
```

**Command or request that led to error**:
```
On rasa shell nlu, enter:
'it was $50.00'

There will be no entities infered for '$' and '50.00'.
```

**Content of configuration file (config.yml)** (if relevant):
```yml
language: en_core_web_md
pipeline:
  - name: SpacyNLP
  - name: SpacyTokenizer
  - name: RegexFeaturizer
  - name: LexicalSyntacticFeaturizer
  - name: CountVectorsFeaturizer
  - name: CountVectorsFeaturizer
    analyzer: "char_wb"
    min_ngram: 1
    max_ngram: 4
  - name: DIETClassifier
    epochs: 100
  - name: EntitySynonymMapper
  - name: ResponseSelector
    epochs: 100
```

**Comment by sara-tagger** [comment]
Thanks for raising this issue, @akelad will get back to you about it soon✨

###### Please also check out the [docs](https://rasa.com/docs/) and the [forum](https://forum.rasa.com/) in case your issue was raised there too 🤗

**Comment by tabergma** [comment]
This should be fixed in our latest version 1.10.3. Can you please try it?

**Comment by JBrettas** [comment]
@tabergma I updated the Rasa version to 1.10.3 and tested with my current design of entities and it's working just fine! Thank you! I consider this issue solved.
